documents/entity/service.go

```diff
@@ -220,3 +220,65 @@ func (s service) DeriveFromUpdatePayload(ctx context.Context, payload *clientent
 
 	return entity, nil
 }
+
+func (s service) GetVersion(ctx context.Context, documentID []byte, version []byte) (documents.Model, error) {
+	return s.get(ctx, documentID, version)
+}
+
+// isDIDaCollaborator returns true if the did is a collaborator of the document
+func (s service) isDIDaCollaborator(ctx context.Context, did identity.DID, model documents.Model) (bool, error) {
+	collAccess, err := model.GetCollaborators()
+	if err != nil {
+		return false, err
+	}
+
+	for _, d := range collAccess.ReadWriteCollaborators {
+		if d == did {
+			return true, nil
+		}
+	}
+	for _, d := range collAccess.ReadCollaborators {
+		if d == did {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (s service) get(ctx context.Context, documentID, version []byte) (documents.Model, error) {
+	selfDID, err := contextutil.AccountDID(ctx)
+	if err != nil {
+		return nil, errors.NewTypedError(documents.ErrDocumentConfigAccountID, err)
+	}
+
+	isCollaborator := false
+	var entity documents.Model
+
+	if s.Service.Exists(ctx, documentID) {
```

Comment:
You dont need to do this check, `GetCurrentVersion` and `GetVersion` should error out if the document is not found.

Reply:
I know but the common case for an entity document is that the document doesn't exist. 

I don't like the pattern

```
 if err != nil {
 //continue with business logic
}
```